src/coreclr/jit/valuenum.cpp

```diff
@@ -10272,7 +10272,7 @@ static bool GetStaticFieldSeqAndAddress(ValueNumStore* vnStore, GenTree* tree, s
     if (vnStore->GetVNFunc(tree->gtVNPair.GetLiberal(), &funcApp) && (funcApp.m_func == VNF_PtrToStatic))
     {
         FieldSeq* fseq = vnStore->FieldSeqVNToFieldSeq(funcApp.m_args[1]);
-        if (fseq->GetKind() == FieldSeq::FieldKind::SimpleStatic)
+        if ((fseq != nullptr) && (fseq->GetKind() == FieldSeq::FieldKind::SimpleStatic))
```

Comment:
We should add `TODO` about fixing the underlying problem here, that being how `VNF_PtrToStatic`s are formed for boxed statics with a temporarily `null` sequence.

Reply:
I can also revert this change, now that I'm actually preserving the field seqs in morph I don't think we hit this case anymore. What would you prefer?

Reply:
I think it's better to leave the check for now. Another case dependent on non-null sequences is guarded by a `noway_assert`, but this one will AV.